packages/application/src/frontend.ts

```diff
@@ -283,6 +283,11 @@ export namespace JupyterFrontEnd {
      */
     readonly currentWidget: Widget | null;
 
+    /**
+     * The user interface mode.
+     */
+    readonly mode: DockPanel.Mode;
```

Comment:
Was this addition intentional?

This means that starting with `^3.1.0-alpha.12`, custom apps implementing a `JupyterFrontEnd.IShell` now need to also implement `mode`.

Is this something we want to keep? It looks like the mode is not used in the context of this PR.

Reply:
This was noticed when updating RetroLab to use the alpha 12 packages: https://github.com/jupyterlab/retrolab/pull/154/commits/7756e96ef86c7211c24a17b30e0b912c4d9171cf